### PR TITLE
Update strings typespecs and add comment

### PR DESCRIPTION
**Pull request by solar05**
Comment inspired by [this commit](https://github.com/elixir-lang/elixir/commit/dd13c508e8cbf4650d3fe5f142980933d5ba8351#diff-fa333eb231fe218a47ae8d74c7ba07c6a1663953e0c9e4ee1c1ee88308f6a872R158)

### CODE REVIEW

File: lib/elixir/lib/string.ex [hunk 1]
```diff
@@ -1902,6 +1903,8 @@ defmodule String do
       "nem rán bERROR! bề"
   """
   @doc since: "1.16.0"
+  @spec replace_invalid(t) :: t
+  @spec replace_invalid(t, binary) :: t
```
Comment (sabiwara) [inline lib/elixir/lib/string.ex:1907]:
```suggestion
  @spec replace_invalid(binary) :: t
  @spec replace_invalid(binary, t) :: t
```

File: lib/elixir/lib/string.ex [hunk 2]
```diff
@@ -1902,6 +1903,8 @@ defmodule String do
       "nem rán bERROR! bề"
   """
   @doc since: "1.16.0"
+  @spec replace_invalid(binary) :: t
```
Comment (josevalim) [inline lib/elixir/lib/string.ex:1906]:
```suggestion
```

File: lib/elixir/lib/string.ex [hunk 3]
```diff
@@ -1845,6 +1845,7 @@ defmodule String do
 
   """
   @spec valid?(t) :: boolean
```
Comment (josevalim) [inline lib/elixir/lib/string.ex:1847]:
```suggestion
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**